src/settings/CredentialsSettings.test.js

```diff
@@ -33,7 +40,59 @@ const renderComponent = () => {
 };
 
 describe('CredentialsSettings', () => {
-  test('rendering CredentialsSettingsForm component', async () => {
+  beforeEach(() => {
+    mockUseQuery.mockReset();
+    mockKyGet.mockReset();
+  });
+
+  test('renders loading state', () => {
+    mockUseQuery.mockReturnValue({
+      data: undefined,
+      error: null,
+      isLoading: true,
+    });
+
+    renderComponent();
+
+    expect(screen.getByText('Icon')).toBeInTheDocument();
+  });
+
+  test('renders error state', () => {
+    mockUseQuery.mockReturnValue({
+      data: undefined,
+      error: { message: 'Network error' },
+      isLoading: false,
+    });
+
+    renderComponent();
+
+    expect(screen.getByText('ui-finc-select.settings.ezbCredentials.error')).toBeInTheDocument();
+  });
+
+  test('renders form when credentials are defined', async () => {
+    mockUseQuery.mockReturnValue({
+      data: {},
+      error: null,
+      isLoading: false,
+    });
+
+    renderComponent();
+
+    await waitFor(() => {
+      expect(screen.getByText('CredentialsSettingsForm')).toBeInTheDocument();
+    });
+  });
+
+  test('renders form when ky.get().json() returns null', async () => {
```

Comment:
A better way of testing would be to use:
```
mockUseQuery.mockReturnValue({
      data: {},
      error: null,
      isLoading: false,
    });
```

And check if the empty text fields will be rendered.

Reply:
This is intentional. It tests that the `queryFn` correctly converts null API responses to `{}` via `res ?? {}`. Your suggestions would skip testing this conversion logic.